pagai/views.py

```diff
@@ -93,6 +89,32 @@ def state(database_name):
         return jsonify({"status": "unknown or training"})
 
 
+@api.route("/explore/<credential_id>/<table>", methods=["GET"])
+def explore(credential_id, table):
+    """
+    Database exploration: returns the first rows of
+    a database table. The db credentials are retrieved from
+    Pyrog. The number of returned rows may be specified using
+    query params (eg: /explore/<db>/<table>?first=10).
+    """
+    credentials = pyrog.get_credentials(credential_id)
+    limit = request.args.get("first", 10, type=int)
+    schema = request.args.get("schema")
+
+    def explore_postgres():
+        connection = postgres.from_pyrog_credentials(credentials)
+        res = postgres.explore(table, connection=connection, limit=limit, schema=schema)
+        connection.close()
+        return res
+
+    try:
+        results = {"POSTGRES": explore_postgres}[credentials.get("model")]()
```

Comment:
Could you explain this line?

Reply:
it's my way of doing a switch in python lol. there is a dict with all possible db models as keys (right now only postgres) and a handler function as value. Then you access the dict value using the actual db model (`dict[credentials.get("model")]`) and if it raises a key error it's like the "default" behavior for a switch. Got it ? How would you write this ?

Reply:
Oh, nice! I think I was confuses because there is only one value for the moment!
Maybe it would be slightly easier to read if you declared the dict outside the `try` and then use it. Having `{}`, `[]`, and `()` on the same line makes it look like a hardcore one-liner ^^